packages/core/src/components/axes/zoom-bar.ts

```diff
@@ -172,21 +172,24 @@ export class ZoomBar extends Component {
 			// update initialZoomDomain and set zoomDomain in model only if the option is changed
 			// not the same object, and both start date and end date are not equal
 			if (
+				newInitialZoomDomain &&
 				!(
-					oldInitialZoomDomain === newInitialZoomDomain ||
-					(oldInitialZoomDomain &&
-						newInitialZoomDomain &&
-						oldInitialZoomDomain[0].valueOf() ===
-							newInitialZoomDomain[0].valueOf() &&
-						oldInitialZoomDomain[1].valueOf() ===
-							newInitialZoomDomain[1].valueOf())
+					oldInitialZoomDomain &&
+					oldInitialZoomDomain[0].valueOf() ===
+						newInitialZoomDomain[0].valueOf() &&
+					oldInitialZoomDomain[1].valueOf() ===
+						newInitialZoomDomain[1].valueOf()
 				)
 			) {
 				this.model.set(
 					{
-						initialZoomDomain: newInitialZoomDomain,
+						// use a new object instead of newInitialZoomDomain
+						initialZoomDomain: Tools.merge(
+							[],
+							newInitialZoomDomain
+						),
 						zoomDomain: newInitialZoomDomain
-							? newInitialZoomDomain
+							? Tools.merge([], newInitialZoomDomain)
```

Comment:
sorry this might be my mistake but does lodash's merge support array merges?

Reply:
could you please verify that this works?

Reply:
Yes, it works as expected.

console.log("newInitialZoomDomain");
console.log(newInitialZoomDomain);
console.log(Tools.merge([], newInitialZoomDomain));
				
![image](https://user-images.githubusercontent.com/59426533/104321447-94dfb200-551e-11eb-82f9-2c409550dbaa.png)

Reply:
👍🏻